src/gil.rs

```diff
@@ -187,42 +176,65 @@ impl ReleasePool {
     }
 }
 
-static mut POOL: *mut ReleasePool = ::std::ptr::null_mut();
+struct ReleasePool {
+    value: UnsafeCell<Option<ReleasePoolImpl>>,
+}
+
+impl ReleasePool {
+    const fn new() -> Self {
+        Self {
+            value: UnsafeCell::new(None),
+        }
+    }
+    unsafe fn get_or_init(&self) -> &mut ReleasePoolImpl {
```

Comment:
It looks like this function isn't thread safe, so I guess you intended for this to be only used alongside the GIL to ensure no race conditions.

```suggestion
    /// # Safety
    /// This function should only be used when the GIL is acquired, to ensure thread safety.
    unsafe fn get_or_init(&self) -> &mut ReleasePoolImpl {
```